libr/util/sys.c

```diff
@@ -1175,6 +1179,20 @@ R_API char *r_sys_pid_to_path(int pid) {
 	if (ret != 0) {
 		return NULL;
 	}
+#elif __HAIKU__
+	char pathbuf[MAXPATHLEN] = {0};
+	int32_t group = 0;
+	image_info ii;
+
+	while (get_next_image_info(0, &group, &ii) == B_OK) {
+		if (ii.type == B_APP_IMAGE) {
+			break;
+		}
+	}
+
+	if (ii.type == B_APP_IMAGE) {
+		memcpy(pathbuf, ii.name, MAXPATHLEN);
```

Comment:
ii.name is a string. use r_str_ncpy() instead of memcpy for proper null termination, avoid unnecessary copies..  (and add a space before (